epic-integration/src/main/kotlin/com/backend/plugins/QuestionnaireCommunication.kt

```diff
@@ -0,0 +1,199 @@
+     * the "Register questionnaire" button you receive the params to send in.
+     * @return id of the created questionnaire or "EMPTY" if a questionnaire was not created.
+     */
+    suspend fun createQuestionnaire(questions: Parameters): String{
+
+        val questionnaire = Questionnaire()
+
+        // The date is set to the current date
+        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+        val dateString = LocalDate.now().format(formatter)
+        val date = SimpleDateFormat("yyyy-MM-dd").parse(dateString)
+
+        questionnaire.setName("NavQuestionnaire")
+        questionnaire.setTitle("Nav questionaire: Funksjonsvurdering")
+        questionnaire.setStatus(Enumerations.PublicationStatus.ACTIVE)
+        questionnaire.setDate(date)
+        questionnaire.setPublisher("NAV")
+        questionnaire.setDescription("Questions about funksjonsvurdering")
+
+        // Set the items for the questionnaire (the questions)
+        var number: Int = 0
+        var items = mutableListOf<Questionnaire.QuestionnaireItemComponent>()
+
+        questions.forEach { t, question ->
+            number++
+            var item = Questionnaire.QuestionnaireItemComponent()
+            item.setLinkId(number.toString())
+            println(question[0])
+            item.setText(question[0])
+            item.setType(Questionnaire.QuestionnaireItemType.STRING)
+            items.add(item)
+        }
+
+        questionnaire.setItem(items)
+
+        // Set the identifier. Should be on the format UUID/patientID.
+        // This allows us to connect a questionnaire to a patient.
+        // TODO: figure out how to search for a questionnaire, this might not work
+        val identifier = Identifier()
+        val uuid = UUID.randomUUID().toString()
+        identifier.setValue("$uuid/1244780")
+        questionnaire.setIdentifier(mutableListOf(identifier))
+
+        val questionnaireJson = jsonParser.encodeResourceToString(questionnaire)
+
+        //post the questionnaire to the server
+        val response: HttpResponse = client.post("$baseURL/Questionnaire"){
+
+            contentType(ContentType.Application.Json)
+            body = questionnaireJson
+        }
+
+        val responseString = response.receive<String>()
+        println("HEADERS: ${response.headers}")
+
+        if (response.headers["Location"] != null) {
+            println(response.headers["Location"])
+            var responseId = response.headers["Location"]!!.split("/")[5]
+            return responseId
+        }
+
+        return "EMPTY"
+    }
+
+    /**
+     * Generates a QuestionnaireResponse to a specific Questionnaire
+     * @param
+     * @param questionnaire Questionnaire the response is related to
+     * @return http response, not QuestionnaireResponse
+     */
+    suspend fun createQuestionnaireResponse(questionnaire: Questionnaire, questionsList: MutableList<String>): HttpResponse {
```

Comment:
Hvis vi først skal lage et system hvor vi har egne klasser (og da også egne filer) for de ulike ressursene, synes jeg vi burde skille mellom Questionnaire og QuestionnaireResponse, da det kan være litt lite intuitivt å lete etter funkjsoner tilknyttet QR i en fil som heter QuestionnaireCommunication.